### PR TITLE
cljfmt: init at 0.11.2

### LINKED ISSUE #242986: Package request: cljfmt
**Project description**
 A tool for formatting Clojure code, used by many editor plugins

**Metadata**

* homepage URL: https://github.com/weavejester/cljfmt
* source URL: https://github.com/weavejester/cljfmt
* license: EPL-1.0
* platforms:  linux, darwin

### PR DESCRIPTION
**Pull request by sg-qwt**
###### Description of changes

A tool for formatting Clojure code. This closes [#242986](https://github.com/NixOS/nixpkgs/issues/242986).

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by nagy — APPROVED** [review]
Code looks good but I did not test the package.

File: pkgs/development/tools/cljfmt/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "-H:Log=registerResource:"
+    "--initialize-at-build-time"
+    "--diagnostics-mode"
+    "--report-unsupported-elements-at-runtime"
+    "--no-fallback"
+  ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
```
Comment (liff) [inline pkgs/development/tools/cljfmt/default.nix:23]: Could also add:

```nix
    mainProgram = "cljfmt";
```

File: pkgs/development/tools/cljfmt/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  extraNativeImageBuildArgs = [
+    "-H:+ReportExceptionStackTraces"
+    "-H:Log=registerResource:"
+    "--initialize-at-build-time"
+    "--diagnostics-mode"
+    "--report-unsupported-elements-at-runtime"
+    "--no-fallback"
+  ];
+
+  passthru.updateScript = nix-update-script { };
```
Comment (liff) [inline pkgs/development/tools/cljfmt/default.nix:21]: Would not hurt to add a simple [package test](https://nixos.org/manual/nixpkgs/stable/#sec-package-tests)?

```nix
  passthru.tests.version = testers.testVersion { package = cljfmt; };
```

**Review by liff — COMMENTED** [review]
The recommended location for new packages is in the [by-name package directory](https://github.com/NixOS/nixpkgs/blob/6c69d3c9caba58c8c9e2362419915a199403bdf6/pkgs/by-name/README.md#limitations).

I don’t think it’s mandatory at this point, though.

**Review by liff — APPROVED** [review]

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 245783` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cljfmt</li>
  </ul>
</details>

Welcome to nixpkgs! The build works and it seems runs fine.

**Pull request merged**